sched/signal/sig_deliver.c

```diff
@@ -182,9 +185,10 @@ void nxsig_deliver(FAR struct tcb_s *stcb)
        * in the current sigprocmask that were already set by newsigprocmask.
        */
 
-      altsigprocmask    = stcb->sigprocmask ^ newsigprocmask;
-      stcb->sigprocmask = (stcb->sigprocmask & altsigprocmask) |
-                          (savesigprocmask & ~altsigprocmask);
+      nxsig_xorset(&tmpset1, &stcb->sigprocmask, &newsigprocmask);
```

Comment:
Maybe we can save some stack by
```suggestion
      nxsig_xorset(&newsigprocmask, &stcb->sigprocmask, &newsigprocmask);
      sigandset(&tmpset1, &stcb->sigprocmask, &tmpset1);
      nxsig_nandset(&tmpset2, &savesigprocmask, &tmpset1);
      sigorset(&stcb->sigprocmask, &tmpset1, &tmpset2);
```
Anyway seems there is a typo as `sigorset` should be done with `tmpset2` and `tmpset3`, but not with `tmpset1` and `tmpset3`

Reply:
That does not look like the same calculation to me because the first nxsig_xorset changes the value of newsigprocmask.  Its it not clear to me that that does not change the value:

      altsigprocmask    = stcb->sigprocmask ^ newsigprocmask;
      stcb->sigprocmask = (stcb->sigprocmask & altsigprocmask) |
                          (savesigprocmask & ~altsigprocmask);

Rearranging and renaming the temporaries to  match the code:

      tmpset1           = stcb->sigprocmask ^ newsigprocmask;
      tmpset2           = stcb->sigprocmask & tmpset1;
      tmpset3           = savesigprocmask & ~tmpset1
      stcb->sigprocmask = tempset2 | tmpset3

You are right!

If it turns out that tmpset1 is equivalent to the use of newsigprocmask, then we could make the other change too.

Reply:
I'll resolve this for now.  Let me know if you prove that the modified newsigprocmask works the same and I will make that change too.